uniqueids/admin.py

```diff
@@ -100,45 +139,61 @@ def save_model(self, request, obj, form, change):
             obj.valid = False
         else:
             for line in rows:
+                if ("address" in line and "address_type" in line):
+                    if (self.validate_address(line)):
+                        existing_address.add(line["address"])
+
                 missing_keys = self.validate_keys(line, obj.import_type)
 
                 if missing_keys:
                     for key in missing_keys:
                         missing_fields.add(key)
-                    obj.valid = False
+
+                missing_keys = self.validate_values(line, obj.import_type)
+
+                if missing_keys:
+                    for key in missing_keys:
+                        invalid_values.add(key)
 
                 if obj.import_type == PersonnelUpload.PERSONNEL_TYPE:
 
                     state = line.get('state')
                     if state and state not in states:
                         missing_states.add(state)
-                        obj.valid = False
 
                     facility = line.get('facility_name')
                     if facility and facility not in facilities:
                         missing_facilities.add(facility)
-                        obj.valid = False
 
                 elif obj.import_type == PersonnelUpload.CORP_TYPE:
                     community = line.get('community')
                     if community and community not in communities:
                         missing_communities.add(community)
-                        obj.valid = False
 
+            if existing_address:
+                errors.append(
+                    "Address invalid or already exists: {}".format(
+                        ', '.join(sorted(existing_address))))
             if missing_fields:
                 errors.append("Missing fields: {}".format(', '.join(
-                    missing_fields)))
+                    sorted(missing_fields))))
+            if missing_keys:
+                errors.append("Missing or invalid values: {}".format(', '.join(
+                    sorted(missing_keys))))
```

Comment:
This should probably be for `missing_fields` and `missing_values`, not `missing_keys` which is a variable that is constantly overwritten in the loop